ocaml/src/osd_bench.ml

```diff
@@ -85,6 +86,92 @@ let sets (client:Osd.osd) progress n value_size period prefix =
   measured_loop progress do_one n >>= fun r ->
   report "sets" r
 
+let range_queries (client:Osd.osd) progress n value_size period prefix =
+  let gen = make_key period prefix in
+  let do_one i =
+    let first_key = gen () in
+    let first = Slice.wrap_string first_key in
+    client # range Osd.High
+           ~first ~finc:false ~last:None ~reverse:false ~max:100
+    >>= fun keys ->
+    Lwt.return ()
+  in
+  measured_loop progress do_one n >>= fun r ->
+  report "ranges" r
+
+let upload_fragments (client:Osd.osd) progress n value_size period prefix =
+  let gen = make_key period prefix in
+  let open Osd_keys in
+  let value = _make_value value_size in
+  let value_slice = Slice.wrap_string value in
+  let namespace_id = 0xffff_ffffl in
+  let namespace_status_key = AlbaInstance.namespace_status ~namespace_id in
+  let active_value = Osd.Osd_namespace_state.(serialize
+                                                to_buffer
+                                                Active)
+                      in
+  let assert_namespace_active =
+    Osd.Assert.value_string
+      namespace_status_key active_value
+  in
+  client # apply_sequence Osd.High []
+         [Osd.Update.set
+            (Slice.wrap_string namespace_status_key)
+            (Slice.wrap_string active_value)
+            Checksum.NoChecksum
+            false;
+         ]
+  >>= maybe_fail
+  >>= fun () ->
+  let recovery_info_slice = _make_value 134 |> Slice.wrap_string in
+  let do_one i =
+    let key = gen () in
+    let object_id = key in
+    let version_id = 0 in
+    let chunk_id = 0 in
+    let fragment_id = 0 in
+    let set_data = Osd.Update.set
+                     (AlbaInstance.fragment
+                        ~namespace_id
+                        ~object_id ~version_id
+                        ~chunk_id ~fragment_id
+                      |> Slice.wrap_string)
+                     value_slice
+                     Checksum.NoChecksum
+                     false
+    in
+    let set_recovery_info =
+      Osd.Update.set
+        (Slice.wrap_string
+           (AlbaInstance.fragment_recovery_info
+              ~namespace_id
+              ~object_id ~version_id
+              ~chunk_id ~fragment_id))
+        recovery_info_slice Checksum.NoChecksum true
+    in
+    let gc_epoch = 0L in
+    let set_gc_tag =
+      Osd.Update.set_string
+        (AlbaInstance.gc_epoch_tag
+           ~namespace_id
+           ~gc_epoch
+           ~object_id ~version_id
+           ~chunk_id ~fragment_id)
+        "" Checksum.NoChecksum true
+    in
+    let asserts = [ assert_namespace_active; ]
+    and updates = [ set_data;
+                   set_recovery_info;
+                   set_gc_tag; ]
+    in
+    client # apply_sequence Osd.High asserts updates >>= maybe_fail
```

Comment:
Reuse Alba_client_upload.upload_packed_fragment_data instead of imitating it here?